oid_validation/pulp/oid_validation/oid_validation.py

```diff
@@ -194,16 +195,18 @@ def _check_extensions(self, cert_pem, dest, log_func, repo_url_prefixes):
 
         valid = False
         for prefix in repo_url_prefixes:
-            # Extract the repo portion of the URL
-            repo_dest = dest[dest.find(prefix) + len(prefix):]
-            try:
-                valid = cert.check_path(repo_dest)
-            except AttributeError:
-                # not an entitlement certificate, so no entitlements
-                log_func('The provided client certificate is not an entitlement certificate.\n')
-            # if we have a valid url check, no need to continue
-            if valid:
-                break
+            if dest.startswith(prefix):
+                # rhsm throws a ValueError if there's no leading /. Amusingly, it immediately
+                # strips it off.
+                repo_dest = '/' + os.path.relpath(dest, prefix)
```

Comment:
I recommend using path manipulation functions whenever possible and avoid string manipulation.  `os.path.join('/', os.path.relpath(dest, prefix))`.  Not a big deal.